tracer_test.go

```diff
@@ -237,6 +237,27 @@ func (s *tracerSuite) TestRandomIDNotZero() {
 	rng.Seed(1) // for test coverage
 }
 
+func (s *tracerSuite) TestReferenceSelfUsesProvidedContext() {
+	ctx := NewSpanContext(
+		TraceID{
+			High: 1,
+			Low:  2,
+		},
+		SpanID(2),
+		SpanID(1),
+		false,
+		nil,
+	)
+	sp1 := s.tracer.StartSpan(
+		"continued_span",
+		opentracing.SpanReference{
+			Type:              SelfRef,
+			ReferencedContext: ctx,
+		},
```

Comment:
just occurred to me, instead of defining a constant of type `opentracing.SpanReferenceType`, we should instead define a factory method 
```go
func SelfRef(ctx jaeger.SpanContext) opentracing.SpanReference
```

this way the reference type (`selfRefType`) constant can be private, and the above call becomes simpler.

Reply:
this also makes it strongly typed, since SelfRef() requires a Jaeger native span context